### PR TITLE
Change: use build-tool for generating dbaas consumers in builds

### PR DESCRIPTION
**Pull request by shreddedbacon**
Use build-deploy-tool to handle the creation of dbaas templates. This replaces the existing BASH process.

At the moment this is replacement of existing functionality, but I think it should also force failure of build on failure to provision dbaas https://github.com/uselagoon/build-deploy-tool/issues/56, but this could be a follow up PR

Sub task of https://github.com/uselagoon/build-deploy-tool/issues/27

### CODE REVIEW

File: internal/generator/services.go [hunk 1]
```diff
@@ -35,6 +35,16 @@ var supportedAutogeneratedTypes = []string{
...
 }
 
+// these are lagoon types that support autogenerated routes
+var supportedDBTypes = []string{
+	"mariadb",
+	"mariadb-dbaas",
+	"postgres",
+	"postgres-dbaas",
+	"mongodb",
+	"mongodb-dbaas",
```
Comment (tobybellwood) [inline internal/generator/services.go:45]: There may also be a "mongo" type as per https://github.com/uselagoon/build-deploy-tool/blob/main/legacy/build-deploy-docker-compose.sh#L558

and this is how we built the internal-services-test (https://github.com/uselagoon/internal-services-test/blob/main/docker-compose.yml#L81)

Comment (shreddedbacon) [reply internal/generator/services.go:45]: :hidethepain:

Comment (tobybellwood) [reply internal/generator/services.go:45]: (confirmed that `lagoon.type: mongo` is the more commonly used option, but it should also work for mongodb too

**Review by tobybellwood — APPROVED** [review]
This is looking good from my side. Tested with all DBaaS providers, in new and update scenarios. The correction of the environment labels is nice, and the addition of the annotations to the postgres consumer both work perfectly. Additionally it clears up a situation where the underlying image for the DBaaS-ed service was being copied into Harbor unnecessarily

Great piece of work. A++

**Pull request merged**